internal/kv/mmdb.go

```diff
@@ -0,0 +1,116 @@
+package kv
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/brexhq/substation/internal/errors"
+	"github.com/brexhq/substation/internal/file"
+	"github.com/oschwald/maxminddb-golang"
+)
+
+// errMMDBKeyMustBeAddr is returned when the key used in a Get call is not a valid
+// IP address.
+var errMMDBKeyMustBeAddr = errors.Error("key must be IP address")
+
+// KvMMDB is a read-only key-value store that is derived from any MaxMind database
+// format (MMDB) file.
+//
+// MMDB is an open source database file format that maps IPv4 and IPv6 addresses to
+// data records, and is most commonly utilized by MaxMind GeoIP databases. Learn more
+// about the file format here: https://maxmind.github.io/MaxMind-DB/.
+type kvMMDB struct {
+	// File contains the location of the MMDB file. This can be either a path on local
+	// disk, an HTTP(S) URL, or an AWS S3 URL.
+	File   string `json:"file"`
+	mu     sync.Mutex
```

Comment:
For a read-heavy KV store like MMDB, should we use a [RWMutex](https://pkg.go.dev/sync#RWMutex) that will only block on mutable calls like `Setup`/`Close` and allow multiple safe `Get` calls?

Reply:
yeah, the IPDB version of this uses RWMutex, so this should too.